samplesheets/tests/test_templatetags.py

```diff
@@ -0,0 +1,227 @@
+
+    def test_get_irods_path_with_project_and_sub_path(self):
+        """Test get_irods_path() with project and sub_path"""
+        project_path = self.irods_backend.get_path(self.project)
+        sub_path = "subfolder1/subfolder2"
+        expected_path = project_path + "/" + sub_path
+        self.assertEqual(
+            s_tags.get_irods_path(self.project, sub_path), expected_path
+        )
+
+    def test_get_irods_path_with_assay_and_sub_path(self):
+        """Test get_irods_path() with assay and sub_path"""
+        assay_path = self.irods_backend.get_path(self.assay)
+        sub_path = "subfolder1/subfolder2"
+        expected_path = assay_path + "/" + sub_path
+        self.assertEqual(
+            s_tags.get_irods_path(self.assay, sub_path), expected_path
+        )
+
+    def test_get_icon_study(self):
+        """Test get_icon() with Study"""
+        icon_html = s_tags.get_icon(self.study)
+        self.assertIn("text-info", icon_html)
+        self.assertIn("mdi:folder-table", icon_html)
+
+    def test_get_icon_assay(self):
+        """Test get_icon() with Assay"""
+        icon_html = s_tags.get_icon(self.assay)
+        self.assertIn("text-danger", icon_html)
+        self.assertIn("mdi:table-large", icon_html)
+
+    def test_get_isatab_tag_html_with_tags_in_TAG_COLORS(self):
+        """Test get_isatab_tag_html() with tags in TAG_COLORS"""
+        isatab = type("MockISATab", (object,), {"tags": TAG_COLORS.keys()})
+        tag_html = s_tags.get_isatab_tag_html(isatab)
+        for tag, color in TAG_COLORS.items():
+            self.assertIn(color, tag_html)
+
+    def test_get_isatab_tag_html_with_unknown_tag(self):
+        """Test get_isatab_tag_html() with an unknown tag"""
+        isatab = type("MockISATab", (object,), {"tags": ["UNKNOWN_TAG"]})
+        tag_html = s_tags.get_isatab_tag_html(isatab)
+        self.assertIn(DEFAULT_TAG_COLOR, tag_html)
+
+    def test_get_request_status_class_with_valid_status(self):
+        """
+        Test get_request_status_class() with values in REQUEST_STATUS_CLASSES
+        """
+        for status, css_class in REQUEST_STATUS_CLASSES.items():
+            irods_request = type(
+                "MockIrodsRequest", (object,), {"status": status}
+            )
+            self.assertEqual(
+                s_tags.get_request_status_class(irods_request), css_class
+            )
+
+    def test_get_request_status_class_with_unknown_status(self):
+        """Test get_request_status_class() with an unknown status"""
+        irods_request = type(
+            "MockIrodsRequest", (object,), {"status": "UNKNOWN"}
+        )
+        self.assertEqual(s_tags.get_request_status_class(irods_request), "")
+
+    def test_trim_base_path(self):
+        """Test trim_base_path() with a realistic iRODS path"""
+        prefix = "/base_path"
+        path = prefix + "/project/subfolder1/subfolder2"
+        expected_trimmed_path = "/project/subfolder1/subfolder2"
```

Comment:
For consistency and clarity, we should call variables like this in tests just `expected`, unless several different variables are needed within one test case.